bugsnag/bugsnagdata_test.go

```diff
@@ -0,0 +1,98 @@
+package bugsnag
+
+import (
+	bugsnaggo "github.com/bugsnag/bugsnag-go"
+)
+
+// bugsnagdata is the utility object which helps with writing bugsnag's unit tests
+type bugsnagdata struct {
+	dataList []interface{}
+}
+
+func newBugsnagData(rawData []interface{}) *bugsnagdata {
+	return &bugsnagdata{
+		dataList: rawData,
+	}
+}
+
+func (dl *bugsnagdata) getBugsnaggoErrorClass() string {
+	for _, data := range dl.dataList {
+		errObj, ok := data.(bugsnaggo.ErrorClass)
+		if ok {
+			return errObj.Name
+		}
+	}
+	return ""
+}
+
+func (dl *bugsnagdata) hasErrTab() bool {
+	for _, data := range dl.dataList {
+		myMap, ok := data.(bugsnaggo.MetaData)
+		if !ok {
+			continue
+		}
+		v, ok := myMap["Error"]
+		if !ok {
+			return false
+		}
+		_, ok = v["details"]
+		if ok {
+			return true
+		}
+		return false
+	}
+	return false
+}
+
+func (dl *bugsnagdata) getLogTab() Rows {
+	for _, data := range dl.dataList {
+		myMap, ok := data.(bugsnaggo.MetaData)
+		if !ok {
+			continue
+		}
+		t, ok := myMap["Log"]
+		return t
+	}
+	return nil
+}
+
+func (dl *bugsnagdata) getContext() string {
+	for _, data := range dl.dataList {
+		cont, ok := data.(bugsnaggo.Context)
+		if ok {
+			return cont.String
+		}
+	}
+	return ""
+}
+
+func (dl *bugsnagdata) getUser() string {
+	for _, data := range dl.dataList {
+		user, ok := data.(bugsnaggo.User)
```

Comment:
could this not just be

```
{
  for _, data := range dl.dataList {
    if user, ok := data.(bugsnaggo.User); ok {
      return user.Id
    }
  }
  return ""
}

Reply:
this was copy pasta oversight.. it is cleaned up